contract/vm.go

```diff
@@ -688,16 +627,13 @@ func (ce *executor) commitCalledContract() error {
 			continue
 		}
 		if v.ctrState != nil {
-			err = state.StageContractState(v.ctrState, bs.StateDB)
+			err = statedb.StageContractState(v.ctrState, bs.StateDB)
 			if err != nil {
 				return newDbSystemError(err)
 			}
 		}
 		/* For Sender */
-		if v.prevState == nil {
-			continue
-		}
```

Comment:
is it OK to remove this check?

Reply:
this logic is ignore putState if state not changed.

I don`t understand why vm.go have their own prevState and curState, so I remove it.

if it needed to ignore PutState about not updated state, it can be added in state.PutState()

Reply:
`prevState` exists for 2 cases: the one above (to avoid unnecessary disk I/O) and the other about revert deploy (remove code from cache)

I am not sure if it is good to remove it

If removed, it must be replaced by equivalent logic

Reply:
you are right. it need to replaced by equivalent logic. it also need to change.

maybe it is good that all AccountState should be exist in state.blockState, not vm.go

Reply:
it is good that all AccountState should be exist in state.blockState, and vm cam get/put accountState from blockState only.

Reply:
Are you sure that the contract state is only modified on "callback"?
I don't even understand what "callback" means on this case
Why a contract state would not be modified on a normal call?